lib/posix/timer.c

```diff
@@ -221,6 +280,15 @@ int timer_delete(timer_t timerid)
 		k_timer_stop(&timer->ztimer);
 	}
 
+	if (timer->sigev_notify == SIGEV_THREAD) {
+		pthread_cancel(timer->thread);
```

Comment:
The spec explicitly says that the threads should be created detached. Therefore we should not call join.